modules/drivers/oracle/test/metabase/driver/oracle_test.clj

```diff
@@ -563,3 +563,42 @@
         (is (=? {:effective_type :type/DateTime}
                 (m/find-first (comp #{"created_at"} :name)
                               (mt/cols (qp/process-query query)))))))))
+
+(deftest date-filter-variable
+  (testing "Date filter variables work against date-times"
+    (mt/test-driver
+      :oracle
+      (mt/dataset
+        date-cols-with-datetime-values
+        (let [query (mt/native-query
+                      {:query "SELECT *
+                               FROM \"mb_test\".\"date_cols_with_datetime_values_dates_with_time\"
+                               WHERE {{date_filter}}"
+                       :template-tags {"date_filter"
+                                       {:name         "date_filter"
+                                        :display-name "Date Filter"
+                                        :type         :dimension
+                                        :dimension    [:field (mt/id :date_cols_with_datetime_values_dates_with_time :date_with_time) nil]
+                                        :widget-type  :date/single}}})
+              query (assoc query :parameters [{:type   :date/single
+                                               :target [:dimension [:template-tag "date_filter"]]
+                                               :value  "2024-11-06"}])]
+          (is (= [[2M "2024-11-06T13:13:13Z"]]
+                 (mt/rows
+                  (qp/process-query query))))
+          (let [query (mt/native-query
+                        {:query "SELECT *
+                               FROM \"mb_test\".\"date_cols_with_datetime_values_dates_with_time\"
+                               WHERE {{date_filter}}"
+                         :template-tags {"date_filter"
+                                         {:name         "date_filter"
+                                          :display-name "Date Filter"
+                                          :type         :dimension
+                                          :dimension    [:field (mt/id :date_cols_with_datetime_values_dates_with_time :date_with_time) nil]
+                                          :widget-type  :date/all-options}}})
+                query (assoc query :parameters [{:type   :date/all-options
+                                                 :target [:dimension [:template-tag "date_filter"]]
+                                                 :value  "2024-11-06"}])]
+            (is (= [[2M "2024-11-06T13:13:13Z"]]
+                   (mt/rows
+                    (qp/process-query query))))))))))
```

Comment:
It looked at first like a single test case is getting duplicated here. There is a difference: the `:widget-type` is either `:date/all-options` or `:date/single`.

I think this would be much easier to read if that difference (and any others; perhaps I missed one) were captured in a `doseq`.

Reply:
You're right! I've fixed this.